src/robot/ConcreteManipulator.cpp

```diff
@@ -0,0 +1,199 @@
+#include "aikido/robot/ConcreteManipulator.hpp"
+#include "aikido/robot/util.hpp"
+
+namespace aikido {
+namespace robot {
+
+//==============================================================================
+ConcreteManipulator::ConcreteManipulator(RobotPtr robot, HandPtr hand)
+  : mRobot(robot), mHand(hand)
+{
+}
+
+//==============================================================================
+HandPtr ConcreteManipulator::getHand()
+{
+  return mHand;
+}
+
+//==============================================================================
+std::unique_ptr<aikido::trajectory::Spline> ConcreteManipulator::smoothPath(
+    const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+    const aikido::trajectory::Trajectory* path,
+    const constraint::TestablePtr& constraint)
+{
+  return mRobot->smoothPath(metaSkeleton, path, constraint);
+}
+
+//==============================================================================
+std::unique_ptr<aikido::trajectory::Spline> ConcreteManipulator::retimePath(
+    const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+    const aikido::trajectory::Trajectory* path)
+{
+  return mRobot->retimePath(metaSkeleton, path);
+}
+
+//==============================================================================
+void ConcreteManipulator::executeTrajectory(
+    const trajectory::TrajectoryPtr& trajectory)
+{
+  mRobot->executeTrajectory(trajectory);
+}
+
+//==============================================================================
+boost::optional<Eigen::VectorXd> ConcreteManipulator::getNamedConfiguration(
+    const std::string& name) const
+{
+  return mRobot->getNamedConfiguration(name);
+}
+
+//==============================================================================
+void ConcreteManipulator::setNamedConfigurations(
+    std::unordered_map<std::string, const Eigen::VectorXd> namedConfigurations)
+{
+  mRobot->setNamedConfigurations(namedConfigurations);
```

Comment:
What if we have a `home` configuration for multiple manipulators? It seems that they might override each other.

Reply:
Hmm. I don't understand. I think this should still work? If a robot has multiple manipulators, it'd do `getManipulator(manipulatorName)->setNamedConfiguration(namedConfiguration)`. Here `mRobot` is not its parent robot but the manipulator's metaSkeleton.